src/ET_Client.php

```diff
@@ -119,6 +118,20 @@ function __construct($getWSDL = false, $debug = false, $params = null)
 					$this->baseSoapUrl = $this->defaultBaseSoapUrl;
 				}
 			}
+			if(array_key_exists('useOAuth2Authentication', $config)){$this->useOAuth2Authentication = $config['useOAuth2Authentication'];}
+			if(array_key_exists("baseUrl", $config))
+			{
+				$this->baseUrl = $config["baseUrl"];
+			}
+			else
+			{
+				if($this->useOAuth2Authentication === false) {
+					throw new Exception("baseUrl is null: Must be provided in config file when instantiating ET_Client");
```

Comment:
do we need this validation as we are getting the base url from token call

Reply:
This is to support not specifying the `baseUrl` in the config file when the legacy authentication is used. For OAuth2 authentication, indeede, the `baseUrl` is set to the value from the /v2/token response.

Reply:
makes sense.